capstone/src/main/java/com/google/sps/servlets/DashboardHandlerServlet.java

```diff
@@ -30,11 +28,8 @@
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         Key userKey = ((User) request.getAttribute(User.USER_ENTITY_NAME)).getUserKey();
-        Filter roomFilter = new FilterPredicate(Room.HOST_PROPERTY_KEY, FilterOperator.EQUAL, userKey);
-        Query query = new Query(Room.ROOM_ENTITY_NAME).setFilter(roomFilter);
-        PreparedQuery results = datastore.prepare(query);
+        PreparedQuery results = FilterService.getQuery(Room.ROOM_ENTITY_NAME, Room.HOST_PROPERTY_KEY, userKey);
```

Comment:
This is really good code de-duplication. I wonder if the helper method currently called "getQuery" could also include turning it into a list like happens below? It seems that is also part of the repeating pattern

Reply:
I would have moved the list into the helper function but Idk how I would get the correct object type to pass in as a parameter? Because the type that the list holds has to be specified, I don't know how I would pass that information over